content/guides/component-testing/framework-configuration.md

```diff
@@ -0,0 +1,417 @@
+---
+title: Framework Configuration
+containerClass: component-testing
+---
+
+Recent years have seen an explosion in component based libraries (Vue, React) and frameworks built on top of them (Nuxt, Next). Cypress tests are written and behave the same regardless. Some frameworks require some additional configuration to work correctly with Cypress component testing.
+
+All the example projects described in this page can be found [here](https://github.com/lmiller1990/cypress-component-examples).
+
+## React (Create React App)
+
+This guide assumes you've created your app using [Create React App](https://create-react-app.dev/docs/documentation-intro). You can find an example project [here](https://github.com/lmiller1990/cypress-component-examples/tree/main/create-react-app).
+
+Once you have a React project, you'll also need to install the Cypress Webpack Dev Server and React adapter, as well as some devDependencies:
+
+```sh
+npm install --save-dev cypress @cypress/vue @cypress/webpack-dev-server html-webpack-plugin@4
+```
+
+<Alert type="info">
+
+<strong class="alert-header">Note</strong>
+
+`html-webpack-plugin@4` is required because the projects created with Create React App use Webpack v4.
+
+</Alert>
+
+Next configure the dev-server to use the same Webpack configuration used by Create React App. We can do this easily using the `react-scripts` plugin provided by Cypress:
+
+```js
+const injectDevServer = require('@cypress/react/plugins/react-scripts')
+
+module.exports = (on, config) => {
+  injectDevServer(on, config)
+  return config
+}
+```
+
+Lastly, tell Cypress where you find your test in `cypress.json`. In this example all the tests are in `src` and named `spec.js`:
+
+```json
+{
+  "testFiles": "**/*.test.{js,ts,jsx,tsx}",
+  "componentFolder": "src"
+}
+```
+
+Finally, add a test. We will replace the default test (using Testing Library) with one using Cypress:
+
+```jsx
+// src/App.test.js
+
+import React from 'react'
+import { mount } from '@cypress/react'
+import App from './App'
+
+it('renders learn react link', () => {
+  mount(<App />)
+  cy.get('a').contains('Learn React')
```

Comment:
Interesting, I did not get this. I will investigate eslint a little more, it would be good if we could do this on our end perhaps, so the user does not need to mess with linting.